core/input/mouse.js

```diff
@@ -108,27 +116,74 @@ Mouse.prototype = {
         this._handleMouseButton(e, 0);
     },
 
+    // Mouse wheel events are sent in steps over VNC. This means that the VNC
+    // protocol can't handle a wheel event with specific distance or speed.
+    // Therefor, if we get a lot of small mouse wheel events we combine them.
+    _generateWheelStepX: function (pos) {
+
+        if (this._accumulatedWheelDeltaX < 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 5);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 5);
+        } else if (this._accumulatedWheelDeltaX > 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 6);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 6);
+        }
+
+        this._accumulatedWheelDeltaX = 0;
+    },
+
+    _generateWheelStepY: function (pos) {
+
+        if (this._accumulatedWheelDeltaY < 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 3);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 3);
+        } else if (this._accumulatedWheelDeltaY > 0) {
+            this._onMouseButton(pos.x, pos.y, 1, 1 << 4);
+            this._onMouseButton(pos.x, pos.y, 0, 1 << 4);
+        }
+
+        this._accumulatedWheelDeltaY = 0;
+    },
+
     _handleMouseWheel: function (e) {
-        if (!this._focused) { return; }
+        if (!this._focused || !this._onMouseButton) { return; }
+
+        window.clearTimeout(this._wheelStepXTimer);
+        window.clearTimeout(this._wheelStepYTimer);
```

Comment:
We also need to clear these timers on "ungrab".

Reply:
Good point!